### ISSUE TITLE
Query result is incorrect when convert to BINARY by CAST

### ISSUE THREAD

**Issue by anitajjx** [matrixorigin/matrixone#832]
#### Can be reproduced ?
Yes

#### Steps:
CREATE TABLE t1 (spID int,userID int,score smallint);
INSERT INTO t1 VALUES (1,1,1);
INSERT INTO t1 VALUES (2,2,2);
INSERT INTO t1 VALUES (2,1,4);
INSERT INTO t1 VALUES (3,3,3);
INSERT INTO t1 VALUES (1,1,5);
INSERT INTO t1 VALUES (4,6,10);
INSERT INTO t1 VALUES (5,11,99);

select CAST(userID AS BINARY) userid_cast from t1;

#### Expected behavior:
Expect to return the following result; if the BINARY type is not supported in current milestone, please issue a proper error message. 
mysql> select CAST(userID AS BINARY) userid_cast from t1;
+--------------------------+
| userid_cast              |
+--------------------------+
| 0x31                     |
| 0x32                     |
| 0x31                     |
| 0x33                     |
| 0x31                     |
| 0x36                     |
| 0x3131                   |
+--------------------------+
7 rows in set (0.01 sec)

#### Actual behavior:

mysql> select CAST(userID AS BINARY) userid_cast from t1;
+-------------+
| userid_cast |
+-------------+
| 1           |
| 2           |
| 1           |
| 3           |
| 1           |
| 6           |
| 11          |
+-------------+
7 rows in set (0.01 sec)

#### Environment:
- Version or commit-id (e.g. v0.1.0 or 8b23a93): commit 3835f50655d37f4174b9bc4ef0d4a871a06f63e4 
- Hardware parameters:
- OS type:
- Others:

#### Additional context:
- Error message from client:
- Server log:
- Other information:

**Comment by broccoliSpicy** [comment]
I think I can fix this in v0.5

**Comment by dengn** [comment]
This feature requires at least BINARY type support. 

The described MySQL behavior is wrong as well. In MySQL, the cast with binary still gives us this result:
```sql
(root@localhost) [mysql_test]>select CAST(userID AS BINARY) userid_cast from t;
+-------------+
| userid_cast |
+-------------+
| 1           |
| 2           |
| 1           |
| 3           |
| 1           |
| 6           |
| 11          |
+-------------+
7 rows in set (0.01 sec)
```

**Comment by dengn** [comment]
#6615 has a feature request for Binary.

**Comment by iceTTTT** [comment]
This bug is fixed after pr #8101 merged.Plz confirm.
